plugins/GSdx/GSSettingsDlg.cpp

```diff
@@ -712,7 +711,16 @@ void GSHacksDlg::OnInit()
 	CheckDlgButton(m_hWnd, IDC_FAST_TC_INV, theApp.GetConfigB("UserHacks_DisablePartialInvalidation"));
 	CheckDlgButton(m_hWnd, IDC_AUTO_FLUSH, theApp.GetConfigB("UserHacks_AutoFlush"));
 	CheckDlgButton(m_hWnd, IDC_UNSCALE_POINT_LINE, theApp.GetConfigB("UserHacks_unscale_point_line"));
-
+	if(ogl && theApp.m_gs_offset_hack.size()==2)
+	{
+		theApp.m_gs_offset_hack.push_back(GSSetting(2,  "Special cases 1", ""));
+		theApp.m_gs_offset_hack.push_back(GSSetting(3,  "Special cases 2", ""));
+	}
+	else if (!ogl && theApp.m_gs_offset_hack.size()==4)
+	{
+		theApp.m_gs_offset_hack.erase(theApp.m_gs_offset_hack.begin() + 2,theApp.m_gs_offset_hack.begin() + 4); //We need to remove element 4 because function deletes elements between first and second including first but excluding second.
+	}
+	ComboBoxInit(IDC_OFFSETHACK, theApp.m_gs_offset_hack, theApp.GetConfigI("UserHacks_HalfPixelOffset"));
```

Comment:
I think you need to clamp here with the index of the last pushed value in the vector, else you'll reach an indeterminate state in the combobobx when you select options exclusive to OpenGL and switch to the Direct3D renderers.

I mean something like,
```
ComboBoxInit(IDC_OFFSETHACK, theApp.m_gs_offset_hack, std::min(max_index_of_combobox, theApp.GetConfigI("UserHacks_HalfPixelOffset")));